### PR TITLE
Update links in data documentation

### PR DESCRIPTION
**Pull request by sjspielman**
This PR addresses issue #1235 and fixes several broken or outdated links in the data documentation. Specifically - 
+ the `kids-first` RNAseq workflow has been deprecated, so links were changed from master to an older commit representing the workflow used for PBTA
+ the manuscript methods section is now  `06` and not `03`.

### CODE REVIEW

File: doc/data-files-description.md [hunk 1]
```diff
@@ -31,66 +31,66 @@ This document contains information about all data files associated with this pro
...
+|`pbta-gene-expression-kallisto.polya.rds` | PBTA data file | [Gene expression abundance estimation](https://github.com/AlexsLemonade/OpenPBTA-manuscript/blob/master/content/06.methods.md#gene-expression-abundance-estimation); [Workflow](https://github.com/kids-first/kf-rnaseq-workflow/blob/052e31844711fa5cd0ef5e581618e9d7bba3fec6/workflow/kfdrc_RNAseq_workflow.cwl) | Gene expression - kallisto TPM for poly-A samples (transcript-level)
+|`pbta-gene-expression-kallisto.stranded.rds` | PBTA data file | [Gene expression abundance estimation](https://github.com/AlexsLemonade/OpenPBTA-manuscript/blob/master/content/06.methods.md#gene-expression-abundance-estimation); [Workflow](https://github.com/kids-first/kf-rnaseq-workflow/blob/052e31844711fa5cd0ef5e581618e9d7bba3fec6/workflow/kfdrc_RNAseq_workflow.cwl)  | Gene expression - kallisto TPM for stranded samples (transcript-level)
```
Comment (jaclyn-taroni) [inline doc/data-files-description.md:47]:
```suggestion
|`pbta-gene-counts-rsem-expected_count.polya.rds` | PBTA data file | [Gene expression abundance estimation](https://github.com/AlexsLemonade/OpenPBTA-manuscript/blob/master/content/06.methods.md#abundance-estimation); [Workflow](https://github.com/kids-first/kf-rnaseq-workflow/blob/052e31844711fa5cd0ef5e581618e9d7bba3fec6/workflow/kfdrc_RNAseq_workflow.cwl) | Gene expression - RSEM expected counts for poly-A samples (gene-level)
|`pbta-gene-counts-rsem-expected_count.stranded.rds` | PBTA data file | [Gene expression abundance estimation](https://github.com/AlexsLemonade/OpenPBTA-manuscript/blob/master/content/06.methods.md#abundance-estimation); [Workflow](https://github.com/kids-first/kf-rnaseq-workflow/blob/052e31844711fa5cd0ef5e581618e9d7bba3fec6/workflow/kfdrc_RNAseq_workflow.cwl)  | Gene expression - RSEM  expected counts for stranded samples (gene-level)
|`pbta-gene-expression-kallisto.polya.rds` | PBTA data file | [Gene expression abundance estimation](https://github.com/AlexsLemonade/OpenPBTA-manuscript/blob/master/content/06.methods.md#abundance-estimation); [Workflow](https://github.com/kids-first/kf-rnaseq-workflow/blob/052e31844711fa5cd0ef5e581618e9d7bba3fec6/workflow/kfdrc_RNAseq_workflow.cwl) | Gene expression - kallisto TPM for poly-A samples (transcript-level)
|`pbta-gene-expression-kallisto.stranded.rds` | PBTA data file | [Gene expression abundance estimation](https://github.com/AlexsLemonade/OpenPBTA-manuscript/blob/master/content/06.methods.md#abundance-estimation); [Workflow](https://github.com/kids-first/kf-rnaseq-workflow/blob/052e31844711fa5cd0ef5e581618e9d7bba3fec6/workflow/kfdrc_RNAseq_workflow.cwl)  | Gene expression - kallisto TPM for stranded samples (transcript-level)
```

File: doc/data-files-description.md [hunk 2]
```diff
@@ -31,66 +31,66 @@ This document contains information about all data files associated with this pro
...
+|`pbta-isoform-expression-rsem-tpm.polya.rds` | PBTA data file | [Gene expression abundance estimation](https://github.com/AlexsLemonade/OpenPBTA-manuscript/blob/master/content/06.methods.md#gene-expression-abundance-estimation); [Workflow](https://github.com/kids-first/kf-rnaseq-workflow/blob/052e31844711fa5cd0ef5e581618e9d7bba3fec6/workflow/kfdrc_RNAseq_workflow.cwl) | Gene expression - RSEM TPM for poly-A samples (transcript-level)
+|`pbta-isoform-expression-rsem-tpm.stranded.rds` | PBTA data file | [Gene expression abundance estimation](https://github.com/AlexsLemonade/OpenPBTA-manuscript/blob/master/content/06.methods.md#gene-expression-abundance-estimation); [Workflow](https://github.com/kids-first/kf-rnaseq-workflow/blob/052e31844711fa5cd0ef5e581618e9d7bba3fec6/workflow/kfdrc_RNAseq_workflow.cwll) | Gene expression - RSEM TPM for stranded samples (transcript-level)
```
Comment (jaclyn-taroni) [inline doc/data-files-description.md:58]: The change in section header need to be addressed here as well.

File: doc/data-files-description.md [hunk 3]
```diff
@@ -31,66 +31,66 @@ This document contains information about all data files associated with this pro
...
+|`tcga-snv-mutation-tmb-coding.tsv` | Analysis file | [`analyses/snv-callers`](https://github.com/AlexsLemonade/OpenPBTA-analysis/tree/master/analyses/snv-callers) | Tumor Mutation burden calculations using coding only mutations identified by both Mutect2 and Strelka2 only within coding sequence regions of the genome.
+|`pbta-star-log-final.tar.gz` | PBTA data file | [Gene expression abundance estimation](https://github.com/AlexsLemonade/OpenPBTA-manuscript/blob/master/content/06.methods.md#gene-expression-abundance-estimation); [Workflow](https://github.com/kids-first/kf-rnaseq-workflow/blob/052e31844711fa5cd0ef5e581618e9d7bba3fec6/workflow/kfdrc_RNAseq_workflow.cwl) | STAR log final output files
+|`pbta-star-log-manifest.tsv` | PBTA data file | [Gene expression abundance estimation](https://github.com/AlexsLemonade/OpenPBTA-manuscript/blob/master/content/06.methods.md#gene-expression-abundance-estimation); [Workflow](https://github.com/kids-first/kf-rnaseq-workflow/blob/052e31844711fa5cd0ef5e581618e9d7bba3fec6/workflow/kfdrc_RNAseq_workflow.cwl) | File to map STAR output to biospecimen IDs
```
Comment (jaclyn-taroni) [inline doc/data-files-description.md:72]: Same comment about section header in the paper

**Review by jaclyn-taroni — COMMENTED** [review]
I caught a few issues related to a change in header in the manuscript for gene expression stuff

File: doc/data-files-description.md (hunk 1 above)
Comment (sjspielman) [reply doc/data-files-description.md:47]: Good finds, thanks!

File: doc/data-files-description.md (hunk 2 above)
Comment (sjspielman) [reply doc/data-files-description.md:58]: `v21` indeed!

**Comment by sjspielman** [comment]
Also noticing it still says current release is v17, so bumping up v21.

**Review by jaclyn-taroni — APPROVED** [review]
It looks like what I caught in my first pass has been addressed. I am approving without going through all the links again.

**Comment by sjspielman** [comment]
I'll give it one more pass before merging, thanks!

**Pull request merged**